internal/utils/miscellaneous.go

```diff
@@ -6,35 +6,32 @@ package utils
 
 import (
 	"fmt"
-	"time"
+	"maps"
+	"strings"
 
-	"github.com/robfig/cron/v3"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 // MergeStringMaps merges the content of the newMaps with the oldMap. If a key already exists then
 // it gets overwritten by the last value with the same key.
 func MergeStringMaps(oldMap map[string]string, newMaps ...map[string]string) map[string]string {
-	var out map[string]string
+	allMaps := []map[string]string{oldMap}
+	allMaps = append(allMaps, newMaps...)
+	return MergeMaps[string, string](allMaps...)
```

Comment:
The compiler performs type inference on generic functions like this from the arguments at compile time, and explicit type annotations like this are completely unnecessary (if not discouraged) in Go.
The function is being called in quite a few places, so I've expanded on this comment in `operator/statefulset/builder.go` through examples.

Reply:
Yes we can change this and remove the explicit types.

Reply:
Addressed in https://github.com/gardener/etcd-druid/pull/777/commits/cd1565084c150999c44c0d592d456fb475a140f2#diff-d730bb6d0ff845d707b521b4a96611c82116adcaa504ed870eeff9534ba3c560